lib/spack/spack/binary_distribution.py

```diff
@@ -1057,7 +1057,10 @@ def generate_package_index(cache_prefix, concurrency=32):
         file_list, read_fn = _spec_files_from_cache(cache_prefix)
     except ListMirrorSpecsError as err:
         tty.error("Unable to generate package index, {0}".format(err))
-        return
+        if required:
```

Comment:
Can you please not handle errors twice... see one line up.

Reply:
so this only gets re-caught in `update_index` for now and the re-raise is so that I can see that the command failed. This is printing an error but then there is nothing communicating to the top level commands that spack failed so it exits 0.

I could raise an empty exception here if that is better?